src/test/rpc/AccountSet_test.cpp

```diff
@@ -203,24 +206,164 @@ class AccountSet_test : public beast::unit_test::suite
 
     void testTransferRate()
     {
```

Comment:
Nice tests. Consider this alternative, that reduces the boilerplate and isolates it away, making the intent clearer, I think: 

```
{
    testcase ("Setting transfer rate (without fix1201)");

    Env env(*this, features(fix1201));
    Account const alice ("alice");
    env.fund(XRP(10000), alice);

    auto test = [&env,&alice](double set, TER code, double get)
    {
        env(rate(alice, set), ter(code));
        BEAST_EXPECT((*env.le(alice))[sfTransferRate] == get * QUALITY_ONE);
    };

    test (0.7, temBAD_TRANSFER_RATE, 1.0);
    test (1.1, tesSUCCESS, 1.1);
    test (1.2, tesSUCCESS, 1.2);
    test (1.0, tesSUCCESS, 1.0);
    test (2.0, tesSUCCESS, 2.0);
    test (2.0, tesSUCCESS, 2.0);
    test (3.1, tesSUCCESS, 3.1);
    test (3.0, tesSUCCESS, 3.0);
    test (1.0, tesSUCCESS, 1.0);
    test (0, tesSUCCESS, 1.0);
    test (1.7, tesSUCCESS, 1.7);
}

{
    testcase ("Setting transfer rate (with fix1201)");

    Env env(*this, features(fix1201));
    Account const alice ("alice");
    env.fund(XRP(10000), alice);
    
    auto test = [&env, &alice](double set, TER code, double get)
    {
        env(rate(alice, set), ter(code));
        BEAST_EXPECT((*env.le(alice))[sfTransferRate] == get * QUALITY_ONE);
    };
    
    test (0.7, temBAD_TRANSFER_RATE, 1.0);
    test (1.1, tesSUCCESS, 1.1);
    test (1.2, tesSUCCESS, 1.2);
    test (1.0, tesSUCCESS, 1.0);
    test (2.0, tesSUCCESS, 2.0);
    test (2.0, tesSUCCESS, 2.0);
    test (3.1, temBAD_TRANSFER_RATE, 2.0);
    test (3.0, temBAD_TRANSFER_RATE, 2.0);
    test (1.0, tesSUCCESS, 1.0);
    test (0, tesSUCCESS, 1.0);
    test (1.7, tesSUCCESS, 1.7);
}
```